server.R

```diff
@@ -155,14 +126,36 @@ shinyServer(function(input, output, session) {
 
   observeEvent(input$update_confirm, {
     if (input$update_confirm == TRUE) {
-      lapply(datatypes, function(x) {
+      lapply(c("project", "template"), function(x) {
         updateSelectInput(session, paste0("dropdown_", x),
           selected = input[[paste0("header_dropdown_", x)]]
         )
       })
     }
   })
 
+  ######## Update Folder List ########
+  lapply(c("header_dropdown_", "dropdown_"), function(x) {
+    observeEvent(ignoreInit = TRUE, input[[paste0(x, "project")]], {
+      # get synID of selected project
+      projectID <- datatype_list$projects[[input[[paste0(x, "project")]]]]
+
+      # gets folders per project
+      folder_list <- synapse_driver$getStorageDatasetsInProject(synStore_obj, projectID) %>% list2Vector()
+
+      # updates foldernames
+      updateSelectInput(session, paste0(x, "folder"), choices = sort(names(folder_list)))
+
+      if (x == "dropdown_") {
+        project_synID <<- projectID
+        datatype_list$folders <<- folder_list
+      }
+
+      req(input$update_confirm)
+      updateSelectInput(session, "dropdown_folder", selected = input[["header_dropdown_folder"]])
+    })
+  })
+
```

Comment:
I have forced it to update folder selection after the folder list updated here

Reply:
@rchai, I see - so lines 154-156 essentially take whatever the user selected in the header tab dropdown, and ensure the 'main' dropdowns (on the first tab of the app) are set to these values?